File: exercism/kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {
  private val sounds = listOf(3 to "Pling", 5 to "Plang", 7 to "Plong")
  fun convert(n: Int): String =
    try {
      sounds.filter { n % it.first == 0 }
                   .map { it.second }
                   .reduce { a, b -> a + b }
    } catch (e: UnsupportedOperationException) {
      n.toString()
    }
}

File: exercism/ruby/raindrops/raindrops.rb
module BookKeeping
  VERSION = 3
end

class Raindrops
  SOUNDS = [["Pling", 3], ["Plang", 5], ["Plong", 7]]

  def Raindrops.convert (n)
    SOUNDS.select do |sound, divider|
      n.modulo(divider).zero?
    end.map(&:first).reduce(:+) || n.to_s
  end
end
